### ISSUE TITLE
Grafana Dashboard home page - keeps reloading in IE9

### ISSUE THREAD

**Issue by sachinmengi** [grafana/grafana#2058]
I have installed Grafana 2.0.2 and its working very well. 

However, there is one specific issue in IE 9, the Dashboard homepage / landing page / login page keeps reloading by default unless and until you click on 'stop' button of the browser. It is a turn-off as the first landing impression goes crazy reloading itself. Can this please be resolved? 

Thanks

**Comment by torkelo** [comment]
what do you mean keeps reloading? does it reload when you go to the login page or when you are on the home page? is it triggered by anything? Do you get the same behavior on http://play.grafana.org  ?

**Comment by sachinmengi** [comment]
I don't see this behavior on play.grafana.org, I think, because it bypass the login page.

It is happening on login page to be precise. It keeps reloading itself, till I click stop of browser(IE 9).

**Comment by torkelo** [comment]
can you login in chrome or firefox?

**Comment by torkelo** [comment]
A similar problem is usually caused by badly configured session store (or out of disk space if you use the file based session store)

**Comment by sachinmengi** [comment]
Yes. Chrome and Firefox works well. However, given that most people in organisation use IE 9, it would not create a good impression.

**Comment by torkelo** [comment]
@sachinmengi I was only asking to exclude causes, can you try to sign on up play.grafana.org and try to login? No one has reported this before so just trying to figure out if it is a bug or a config issue

**Comment by sachinmengi** [comment]
play.grafana.com - login page is working fine on IE 9. However, when I put http://play.grafana.org/login into my browser, it makes it http://play.grafana.org/#/login. Note the # in the URL.

However, when I try my grafana instance, it keep the URL static i.e no # is inserted. http://<servername>:3000/login and keeps reloading.

**Comment by sachinmengi** [comment]
Also, in terms of storage, following is what I have

#################################### Paths ####################################
[paths]
# Path to where grafana can store temp files, sessions, and the sqlite3 db (if that is useD)
#
;data = /var/lib/grafana
#
# Directory where grafana can store logs
#
;logs = /var/log/grafana

Also, space is not a problem
Filesystem           1K-blocks      Used Available Use% Mounted on
/dev/mapper/VolGroup00-LogVol00
                      91510768  15548224  71239040  18% /

**Comment by sachinmengi** [comment]
I am running on 2.0.2 version and play site is running on 2.0.3. Do you think that would make any difference depending on the changes?

**Comment by sachinmengi** [comment]
I have reinstalled 2.0.2 in a different infrastructure to see if the issue persists. On IE9, the issue shows up.

**Comment by sachinmengi** [comment]
Also, are you using Angular JS as part of the server. There seems to be a known issue with IE9 and Angular JS
https://github.com/angular/angular.js/issues/9235

**Comment by torkelo** [comment]
@sachinmengi angularjs is not part of the server, it is only used in the frontend (as it is browser js lib). You might be right seems to be related to that angularjs bug. Strange that you cant replicate on play.grafana.org

**Comment by sachinmengi** [comment]
Have you changed / upgraded front end component using Angular JS as part of 2.0.3 vs. 2.0.2?

Play site is running 2.0.3-pre1 where as I am running 2.0.2.
Can you make 2.0.3 download available for me to test and confirm if behavior is fixed?

**Comment by torkelo** [comment]
http://grafana.org/download/builds.html

**Comment by sachinmengi** [comment]
I have tried other version and the issue remains. I found this article, where the same problem is faced but (outside grafana) and fixed using server side redirect rules. http://stackoverflow.com/questions/19276781/angularjs-infinite-loop-when-redirecting/26457731#26457731

For me also, the URL keeps getting appended with login endlessly:
http://local.grafana.org:3000/login#/login#%2Flogin%23%2Flogin....

Now I believe, this is not a code issue and also not installation issue. This is related to how server URL settings may have been setup which triggers IE9 behavior. Can you share any such settings which might have been done on play site?

**Comment by torkelo** [comment]
Its behind nginx: 

```
 server {
    listen 80;
    server_name play.grafana.org;

    location / {
      proxy_pass                 http://127.0.0.1:3000/;
  }
}
```

and in grafana.ini root url is:
```ini
[server]
domain = play.grafana.org
root_url = http://play.grafana.org
http_port = 3000
```

**Comment by sachinmengi** [comment]
No luck. Are we missing any other settings. Coz , only on IE, URLs get redirected with #

1) http://play.grafana.org   --> http://play.grafana.org/#/
2) http://play.grafana.org/login  --> http://play.grafana.org/#/login

**Comment by torkelo** [comment]
@sachinmengi the # is becuase it does not support html5 and pushstate , I cant think of anything else

**Comment by sachinmengi** [comment]
I am not getting anywhere with it. Lot of such known issues on Angular JS and IE9 compatability (i know it still doesn't explain why play site doesn't have these issues) which were fixed in 1.3 version of Angular JS. Do you have any plans to upgrade to newer version of Angular JS?

https://github.com/angular/angular.js/issues/6820
https://github.com/angular/angular.js/issues/9380

**Comment by torkelo** [comment]
Grafana v2.0.2 is already running Angular 1.3 , plan to upgrade to 1.4 (just released) soon (before 2.1 release)

**Comment by sachinmengi** [comment]
ok thanks. 

grafana-master\public\vendor , license.json shows following 
  "angular": {
    "version":"1.1.5",
    "license":"MIT"
  },
  "angular-dragdrop": {
    "version":"1.0.4",
    "license":"MIT"
  }, hence I asked.

**Comment by torkelo** [comment]
that file is outdated

**Comment by sachinmengi** [comment]
yes, thanks. I can see in the code, its using v1.3.4 at the moment.

**Comment by sachinmengi** [comment]
Would it be possible for you to have 1.3.15 version (the latest of 13) and provide me with a build to just test it.

**Comment by sachinmengi** [comment]
can you please share your nginx.conf settings (full file), just so that I compare with mine.

**Comment by damm** [comment]
I'll share mine for posterity; I run Grafana2 under `/grafana2` so there's a slight change to the config setting.

```
    location /grafana2/ {
        port_in_redirect off;
        proxy_read_timeout 90;
        proxy_pass http://localhost:4000/;
        proxy_set_header Host $host;
    }
```

then in grafana.ini I have
```
root_url = %(protocol)s://%(domain)s:%(http_port)s/grafana2/
```

P.S. Don't forget to check out the Web Browser Console for error messages

**Comment by Convolution** [comment]
Has anyone found a solution for this yet?

I have tested this on multiple systems and multiple servers but every time I try to use IE9 (Default browser at my company) the login page keeps refreshing and will eventually crash the browser. Chrome and Firefox work just fine.

The problem only manifests itself on the login page. If I load the page in IE9 and repeatedly click the 'stop' button, the refreshing stops and I can login normally. After that Grafana works just fine. 

I've also tried to use the sign in feature on play.grafana.org. This works but also has a quirck. I initially get the modified login page with the 'sign in with google/github' buttons. Then after a couple of seconds it refreshes and the google/github buttons are gone. It refreshes only once and works just fine after that.

The problem is very easy to reproduce, Just host a grafana server and connect to it directly with IE9. (Default or compatibility mode)

Version string: Grafana version: 2.0.2, commit: v2.0.2, build date: 2015-04-22 17:51:5

**Comment by sachinmengi** [comment]
Nope, I couldn't and excluded IE9 from browsers supported in my project. How play site is working is still a mystery for me ..

**Comment by PaulKuiper** [comment]
The same is happening for us. It keeps reloading. The standard browser is IE9 for our company ;-(.

**Comment by ymettier** [comment]
Hello,

Same for me.
Grafana version: 2.1.3, commit: v2.1.3, build date: 2015-08-24 10:02:44
(from rpm, running on Redhat 6)

And...
F12 - Console

Lots of this :
```
HTML1113: Document mode restart from Quirks to IE9 Standards 
login
```

Sometimes this :
```
HTML1113: Document mode restart from Quirks to IE9 Standards 
login
HTML1115: X-UA-Compatible META tag ('IE=edge,chrome=1') ignored because document mode is already finalized. 
login
```

(where `login` is a link, e.g. blue and underlined)


About http://play.grafana.org (which is `Grafana version: 2.1.0-pre1, commit: v2.0.2+534-g8daec7b, build date: 2015-07-15 14:57:52`)

* http://play.grafana.org/ is working (redirects on http://play.grafana.org/#/)
* http://play.grafana.org/login is... well, it shows the ability to log in with google or github, but does not let me time to do anything : it redirects immediately on http://play.grafana.org/#/login with no possibility to log in with google or github.

And that let me think that I have LDAP authentication enabled (and working with Firefox).
Is there a problem with IE9 and the authentication system ?

Regards,
Yves

**Comment by eyinsma** [comment]
Same issue when try latest Grafana v2.6 on IE9. Any workaround?

**Comment by xiaoping378** [comment]
Now, the latest master has the same issue.

![image](https://cloud.githubusercontent.com/assets/5623173/13342926/9ce63dc2-dc81-11e5-933d-b96b48576015.png)

**Comment by torkelo** [comment]
IE9 is not supported , any chance you can use a modern browser ?

**Comment by PaulKuiper** [comment]
Hi Torkelo,

We have luckely moved to IE11 and I always advise the users to use Chrome or Firefox. So this is solved now. Thanks very much.
